### DISCUSSION TITLE
如何用代码调用本脚本功能

### DISCUSSION THREAD

**Discussion by Ethkuil** [the1812/Bilibili-Evolved#3707] [提问 / Q&A] [answered]
以下载弹幕为例，目前我只会打开视频后再点击GUI下载。我想下载几百多个视频的弹幕，这就变得非常困难。

我目前也想到了一些可能的解决方案，但不确定是否合理可行。

想法1. 用Selenium逐个打开视频页面，模拟手工点击Bilibili-Evolved的GUI

想法2. 我找到了下载弹幕组件的代码链接[danmaku/download.js](https://raw.githubusercontent.com/the1812/Bilibili-Evolved/master/registry/dist/components/video/danmaku/download.js)，也许可从中提取出下载用的代码？（我还没把代码复制到本地格式化，所以还没看懂）

**Comment by timongh** [comment, marked as answer]
也可以用原始接口写：[bilibili-API-collect](https://github.com/SocialSisterYi/bilibili-API-collect)

**Reply by Ethkuil** [reply]
真好用，备份评论区的需求也解决了~

**Reply by timongh** [reply]
刚刚没注意，你提到的 `danmaku/download.js` 不是源码，是组件的编译产物。组件源码文件夹在[这里](https://github.com/the1812/Bilibili-Evolved/blob/master/registry/lib/components/video/danmaku/download)。组件的组织结构见这里：[CONTRIBUTING](https://github.com/the1812/Bilibili-Evolved/blob/master/CONTRIBUTING.md)
